app/views/handlers/confirm_email.py

```diff
@@ -108,43 +110,51 @@ def question_schema(self) -> QuestionSchemaType:
         }
 
     @cached_property
-    def is_email_correct(self):
+    def is_email_correct(self) -> bool:
         return self._form_data.get("confirm-email") == CONFIRM_EMAIL_YES_VALUE
 
-    def get_context(self):
+    def get_context(self) -> dict:
         return build_confirm_email_context(self.question_schema, self.form)
 
-    def get_next_location_url(self):
+    def get_next_location_url(self) -> str:
         if self.is_email_correct:
             return url_for(
                 ".get_confirmation_email_sent",
                 email=self._serialized_email,
             )
         return url_for(".send_confirmation_email", email=self._serialized_email)
 
-    def get_page_title(self):
+    def get_page_title(self) -> str:
         # pylint: disable=no-member
         # wtforms Form parents are not discoverable in the 2.3.3 implementation
         if self.form.errors:
-            return gettext("Error: {page_title}").format(page_title=self.page_title)
+            formatted_errors: str = gettext("Error: {page_title}").format(
+                page_title=self.page_title
+            )
+            return formatted_errors
         return self.page_title
 
-    def handle_post(self):
+    def handle_post(self) -> None:
         if self.is_email_correct:
             self._publish_fulfilment_request()
-            self._session_store.session_data.confirmation_email_count += 1
+            # type ignore: session data would be populated at this point
+            self._session_store.session_data.confirmation_email_count += 1  # type: ignore
             self._session_store.save()
 
-    def _publish_fulfilment_request(self):
+    def _publish_fulfilment_request(self) -> Task:
         fulfilment_request = ConfirmationEmailFulfilmentRequest(
             self._email,
-            self._session_store.session_data,
-            self._questionnaire_store.metadata,
+            # type ignore: session data would be populated at this point
+            self._session_store.session_data,  # type: ignore
+            # type ignore: metadata will be populated as we reach this stage
+            self._questionnaire_store.metadata,  # type: ignore
             self._schema,
         )
 
         try:
-            return current_app.eq["cloud_tasks"].create_task(
+            # type ignore: mypy not aware of eq attribute but it is a cloud task publisher
+            cloud_task_publisher: CloudTaskPublisher = current_app.eq["cloud_tasks"]  # type: ignore
```

Comment:
I've just realised this could be a `CloudTaskPublisher` or a `LogCloudTaskPublisher` see the setup.py, theres two potential options around line 370, I know we looked at this one, missed that 😁

Reply:
I think it's CloudTaskPublisher as its calling Task from app.cloud_tasks.cloud_task_publishers.CloudTaskPublisher .